pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -324,17 +332,9 @@ func (c *ConnectedObjectsQuery) reverseExpandTupleToUserset(
 	targetObjectType := req.targetObjectRef.GetType()
 	targetObjectRel := req.targetObjectRef.GetRelation()
 
-	tuplesetRelation := req.ingress.TuplesetRelation.GetRelation()
-
 	var userFilter []*openfgav1.ObjectRelation
 
-	// e.g. 'user:bob'
-	if val, ok := req.sourceUserRef.(*UserRefObject); ok {
-		userFilter = append(userFilter, &openfgav1.ObjectRelation{
-			Object: tuple.BuildObject(val.Object.Type, val.Object.Id),
-		})
-	}
-
```

Comment:
This is dead code that can never be hit.

Reply:
@miparnisari I'm wondering if we should just `panic` in these cases. We don't ever expect to hit the code paths, but if we do it's worth panic'ing to bring attention to the invariant that was not met.

Instead of swallowing the issue and continuing on, can we panic and then just codecov ignore the blocks of code? That way we're not hurting our codecov but we're also catching the invariant problem if the code path gets hit someone because of a problem introduced somewhere.

Reply:
> can we panic and then just codecov ignore the blocks of code?

No, I don't like the idea of introducing a panic. It will never hit and it'll make the code just a tiny bit longer to read. 

If we could somehow introduce a test for this, it will suffice. But we can't, because we can never write a model where the invariant is broken 😞

Reply:
> No, I don't like the idea of introducing a panic.

Why? 


> If we could somehow introduce a test for this, it will suffice. But we can't (I think), because we can never write a model where the invariant is broken 😞

This is exactly why `panic` is warranted IMO. If you cannot add a test for it because it is an invariant that is always expected to be met, then you should be screaming an error if/when the invariant is met.

Reply:
Ok, added the panic